pytorch_lightning/trainer/trainer.py

```diff
@@ -509,16 +509,14 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
                 self.state.status = TrainerStatus.INTERRUPTED
                 self.on_keyboard_interrupt()
                 # same treatment as below
-                if not isinstance(self.training_type_plugin, DDPSpawnPlugin):
-                    self.accelerator.teardown()
+                self.accelerator.teardown()
         except BaseException:
             self.state.status = TrainerStatus.INTERRUPTED
             if distributed_available() and self.world_size > 1:
                 # try syncing remaing processes, kill otherwise
                 self.training_type_plugin.reconciliate_processes(traceback.format_exc())
             # give accelerators a chance to finish
-            if not isinstance(self.training_type_plugin, DDPSpawnPlugin):
-                self.accelerator.teardown()
+            self.accelerator.teardown()
```

Comment:
@SeanNaren - do you see us needing to call accelerator's teardown in exceptional cases? i feel this could have more side effects or even create other exceptions which will mask the original errors. if possible, do you think we should remove this teardown call entirely?